### PR TITLE
Feat/add error and product status information to the mailpoet marketing channel - MAILPOET-5696

### PR DESCRIPTION
**Pull request by triple0t**
## Description

Add error and product status information to the mailpoet marketing channel.

This PR handles the implementation of `get_product_listings_status` and `get_errors_count`

This PR is concerned with showing the MailPoet sending status. 
i.e, We show sending error or error count related to sending issues

MSS: All good, no issues
Screenshot 2023-11-28 at 11 53 34 AM

MSS: Sending paused issue to resolve
Screenshot 2023-11-27 at 12 28 48 PM

MSS: Sending okay, issue to resolve 
Screenshot 2023-11-27 at 12 38 17 PM

Other sending methods, no issues
Screenshot 2023-11-28 at 11 55 14 AM

Other sending methods with issues
Screenshot 2023-11-28 at 9 56 33 AM

## Code review notes

Please check commits

## QA notes

Start with the QA notes here: https://github.com/mailpoet/mailpoet/pull/5289
Initiate an error e.g authorized email address error, check the WooCommerce Marketing tab at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5696](https://mailpoet.atlassian.net/browse/MAILPOET-5696)

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5696]: https://mailpoet.atlassian.net/browse/MAILPOET-5696?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/WooCommerce/MultichannelMarketing/MPMarketingChannel.php [hunk 1]
```diff
@@ -131,4 +158,13 @@ protected function isMPSetupComplete(): bool {
...
   }
+
+  /**
+   * Check if user is sending with the MailPoet sending method
+   * @return bool
+   */
+  protected function isUserSendingWithMSS(): bool {
+    $sendingMethod = $this->settings->get('mta.method', SettingsController::DEFAULT_SENDING_METHOD);
+    return $sendingMethod === Mailer::METHOD_MAILPOET;
+  }
```
Comment (costasovo) [inline mailpoet/lib/WooCommerce/MultichannelMarketing/MPMarketingChannel.php]: I would suggest replacing this method with[ `Bridge::isMailpoetSendingServiceEnabled`.](https://github.com/mailpoet/mailpoet/blob/b4b49166f24f2b6412b1374e82ac9ccdc2eb7521/mailpoet/lib/Services/Bridge.php#L79-L87) The logic is not very complex, but I think we should reuse existing code when possible.

**Review by costasovo — COMMENTED** [review]
@triple0t Great job! Looks good. I suggest reusing existing code for checking if user uses MSS. See my other comment.

File: mailpoet/lib/WooCommerce/MultichannelMarketing/MPMarketingChannel.php (hunk 1 above)
Comment (triple0t) [reply mailpoet/lib/WooCommerce/MultichannelMarketing/MPMarketingChannel.php]: Thank you for the suggestion. Implemented in https://github.com/mailpoet/mailpoet/pull/5312/commits/a5a022c130346b36d413cccf9c5119104db45ddb

**Comment by triple0t** [comment]
Thank you for the review, @costasovo.

I've implemented your suggestion.

**Review by costasovo — APPROVED** [review]
Thank you for refactoring the MSS check! 👍

**Comment by Aschepikov** [comment]
Hi @triple0t when the key is incorrect and the sending is paused I don't see the error on the channel info
image
When I have an error with another sending method I don't see the issue, is it expected?
image

**Comment by triple0t** [comment]
Hi @Aschepikov 

>  when the key is incorrect and the sending is paused I don't see the error on the channel info

> When I have an error with another sending method I don't see the issue, is it expected?

We are showing **Sending** error on the channel status.

You can try sending a Newsletter. The error should be logged and you will see the status on the multichannel dashboard page.

**Comment by Aschepikov** [comment]
@triple0t  I see it, thank you!

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**